src/main/java/ch/njol/skript/expressions/ExprIP.java

```diff
@@ -21,61 +21,96 @@
 
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
-
-import org.bukkit.entity.Player;
-import org.bukkit.event.Event;
-import org.bukkit.event.player.PlayerLoginEvent;
-import org.eclipse.jdt.annotation.Nullable;
-
-import ch.njol.skript.ScriptLoader;
 import ch.njol.skript.Skript;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
-import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
-import ch.njol.skript.lang.SkriptParser.ParseResult;
-import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.ScriptLoader;
+import ch.njol.util.coll.CollectionUtils;
 import ch.njol.util.Kleenean;
+import com.destroystokyo.paper.event.server.PaperServerListPingEvent;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerLoginEvent;
+import org.bukkit.event.server.ServerListPingEvent;
+import org.eclipse.jdt.annotation.Nullable;
 
 @Name("IP")
-@Description("The IP address of a player.")
+@Description({"The IP address of a player, or the connected player in a <a href='events.html#connect'>connect</a>> event, " +
+		"or the pinger in a <a href='events.html#server_list_ping'>server list ping</a> event."})
 @Examples({"IP-ban the player # is equal to the next line",
 		"ban the IP-address of the player",
-		"broadcast \"Banned the IP %IP of player%\""})
-@Since("1.4, 2.2-dev26 (when used in connect event)")
+		"broadcast \"Banned the IP %IP of player%\"",
+		"",
+		"on connect:",
+		"	broadcast \"%player%: %ip%\"",
+		"",
+		"on server list ping:",
+		"	send \"%ip adress of the pinger%\" to the console"})
+@Since("1.4, 2.2-dev26 (when used in connect event), INSERT VERSION (server ping event and pattern improvements)")

```

Comment:
remove the pattern improvements part, it's useless because you can't tell what patterns changed by just looking at the docs